docs/encoding.md

```diff
@@ -0,0 +1,327 @@
+easily (PowerShell's encoding settings are fragmented, and can be changed back at any time in the Integrated Console).
+
+These factors mean the PowerShell extension relies on the user to configure the encoding instead.
+
+## Choosing the right encoding
+
+To solve encoding issues, you need to choose a common text encoding that
+you can encode all your PowerShell scripts in that everything you use will understand.
+
+**The encoding most likely to work easily with VSCode, all versions of PowerShell
+and most (especially Windows-based) applications is UTF-8 with a [byte-order mark] (BOM)**.
+
+**However**, choosing an encoding is a question of what platforms and applications
+will be reading and writing your text/PowerShell scripts.
+
+On Windows, many applications have long used [Windows-1252],
+although many .NET applications use [UTF-16]
+(the Windows world often calls this "Unicode", a term that now [refers to a broader standard](https://en.wikipedia.org/wiki/Unicode)).
+
+In the Linux world, on the web, and [where .NET is heading](https://github.com/dotnet/standard/issues/260#issuecomment-289549508)
+[UTF-8] is now the dominant encoding.
+
+Unicode encodings (the "UTF"s) also have the concept of a [byte-order mark] (BOM),
+which may occur at the beginning of text to
+tell a decoder what encoding the text is in and, in the case of
+multibyte encodings, the [endianness](https://en.wikipedia.org/wiki/Endianness) of the encoding.
+BOMs are also designed to be bytes that rarely occur in non-Unicode text,
+allowing a reasonable guess that text is Unicode when a BOM is present.
+
+BOMs are optional and their adoption has not caught on in the Linux world,
+due to a dependable convention of UTF-8 being used everywhere.
+This means that most Linux applications presume that text input is encoded in UTF-8.
+While many Linux applications will recognise and correctly handle a BOM,
+a number do not, leading to artifacts in text manipulated with those applications.
+
+**Therefore**:
+
+- If you work primarily with Windows applications and Windows PowerShell,
+  you should prefer an encoding like UTF-8 with BOM or UTF-16.
+- If you work across platforms, you should prefer UTF-8 with BOM.
+- If you work mainly in Linux-associated contexts, you should prefer UTF-8 without BOM.
+- Windows-1252 and latin-1 are essentially legacy encodings that you should avoid if possible.
+  However, some older Windows applications may depend on them.
+- It's also worth noting that script signing [is encoding-dependent](https://github.com/PowerShell/PowerShell/issues/3466),
+  meaning a change of encoding on a signed script will require resigning.
+
+## Configuring VSCode
+
+VSCode's default encoding is UTF-8 without BOM.
+
+To set [VSCode's encoding](https://code.visualstudio.com/docs/editor/codebasics#_file-encoding-support),
+go to the VSCode settings (<kbd>Ctrl</kbd>+<kbd>,</kbd>)
+and set the `"files.encoding"` setting:
+
+```json
+"files.encoding": "utf8bom"
```

Comment:
It may be worth pointing out that you can make this encoding change *just* for PoweShell files:
```
    "[powershell]": {
        "files.encoding": "utf8bom",
        "files.trimTrailingWhitespace": true
    }
```